.github/workflows/npm-beta-release.yml

```diff
@@ -0,0 +1,117 @@
+name: NPM Beta Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      snapshot_tag:
+        description: 'NPM dist-tag for the beta release'
+        required: true
+        default: 'beta'
+        type: choice
+        options:
+          - beta
+          - alpha
+          - rc
+          - preview
+
+concurrency:
+  group: npm-beta-release
+  cancel-in-progress: false
+
+env:
+  LOG_FORMAT: PRETTY
+  TURBO_TELEMETRY_DISABLED: 1
+  TURBO_API: https://cache.depot.dev
+  TURBO_TOKEN: ${{ secrets.DEPOT_TURBO_TOKEN }}
+  TURBO_TEAM: ${{ secrets.DEPOT_ORG_ID }}
+
+jobs:
+  beta-release:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v6
+        with:
+          node-version-file: .nvmrc
+
+      - name: Get short SHA
+        id: sha
+        run: echo "short=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
+
+      - name: Get base version
+        id: version
+        run: |
+          BASE_VERSION=$(node -p "require('./typescript/sdk/package.json').version")
+          BETA_VERSION="${BASE_VERSION}-${{ inputs.snapshot_tag }}.${{ steps.sha.outputs.short }}"
+          echo "base=$BASE_VERSION" >> $GITHUB_OUTPUT
+          echo "beta=$BETA_VERSION" >> $GITHUB_OUTPUT
+
+      - name: Check for changesets
+        id: changesets
+        run: |
+          CHANGESET_COUNT=$(find .changeset -name "*.md" ! -name "README.md" ! -name "config.json" 2>/dev/null | wc -l | tr -d ' ')
+          echo "count=$CHANGESET_COUNT" >> $GITHUB_OUTPUT
+          if [ "$CHANGESET_COUNT" -eq 0 ]; then
+            echo "::warning::No pending changesets found. Beta releases require pending changeset files."
+            echo "To create a beta release, first add a changeset with: yarn changeset"
+          else
+            echo "Found $CHANGESET_COUNT changeset(s)"
+          fi
+
+      - name: Install dependencies
+        if: steps.changesets.outputs.count != '0'
+        run: yarn install --immutable
+
+      - name: Create snapshot versions
+        if: steps.changesets.outputs.count != '0'
+        run: yarn changeset version --snapshot ${{ inputs.snapshot_tag || 'beta' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build packages
+        if: steps.changesets.outputs.count != '0'
+        run: yarn build
+
+      - name: Publish beta packages
+        if: steps.changesets.outputs.count != '0'
+        run: |
+          echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" >> $HOME/.npmrc
+          yarn changeset publish --tag ${{ inputs.snapshot_tag || 'beta' }}
```

Comment:
Consider `--no-git-tag` maybe?
See: https://github.com/changesets/changesets/blob/main/docs/snapshot-releases.md#using-the---no-git-tag-flag